gym/spaces/discrete.py

```diff
@@ -23,6 +23,24 @@ def contains(self, x):
             return False
         return as_int >= 0 and as_int < self.n
     def __repr__(self):
-        return "Discrete(%d)" % self.n
+        return "Discrete({})".format(self.n)
     def __eq__(self, other):
         return self.n == other.n
+
+class Categorical(Discrete):
+    """A discrete space with named elements. Element indices are made available
+    as attributes (as long as they're legal Python names). e.g.
+    >>> shapes = Categorical(['circle', 'triangle', 'rectangle'])
+    >>> shapes.circle
+    0
+    """
+    def __init__(self, named_members):
+        self.n = len(named_members)
+        self.named_members = list(named_members)
+    def __repr__(self):
+        return "Categorical({!r})".format(self.named_members)
+    def __getattr__(self, attr):
```

Comment:
Done in [7edd7ea](https://github.com/openai/gym/pull/389/commits/7edd7ea9fe44fcef748f50c8801f38987ca3ce5f). Another option would be to use `__getitem__`, which would remove any concerns about namespace pollution, but require a few more keystrokes to invoke.